rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -118,18 +104,31 @@ public boolean isBookKeeperAlive()
   }
 
   @VisibleForTesting
-  BookKeeper(Configuration conf, MetricRegistry metrics, Ticker ticker) throws FileNotFoundException
+  BookKeeper(Configuration conf, MetricRegistry metrics, Ticker ticker) throws BookKeeperInitializationException
   {
     this.conf = conf;
     this.metrics = metrics;
     this.ticker = ticker;
+    this.splitSize = CacheConfig.getCacheFileSplitSize(conf);
+    cleanupOldCacheFiles(conf);
+
+    setupCacheDirectory(conf);
     initializeMetrics();
     initializeCache(conf, ticker);
-    cleanupOldCacheFiles(conf);
     fetchProcessor = new RemoteFetchProcessor(this, metrics, conf);
     fetchProcessor.startAsync();
   }
 
+  private void setupCacheDirectory(Configuration conf) throws BookKeeperInitializationException
+  {
+    try {
+      CacheUtil.createCacheDirectories(conf);
```

Comment:
With this change, we can remove creating the directories from bookkeeper startup script.

Reply:
This change is a reordering and does not change any functionality (was previously found in `initializeCache()`).
We still need the parent directories created through the BK startup script in order to create the cache directories.